addon/modifiers/did-resize.js

```diff
@@ -6,23 +6,41 @@ export default class DidResizeModifier extends Modifier {
   options = {};
 
   // Private API
-  observer = null;
+  _didInstall = false;
+  static observer = null;
+  static handlers = [];
+
+  addHandler() {
+    DidResizeModifier.handlers.push({
+      element: this.element,
+      handler: this.handler,
+    });
+  }
+
+  removeHandler() {
+    DidResizeModifier.handlers.splice(
+      DidResizeModifier.handlers.findIndex((h) => h.element === this.element),
+      1
+    );
+  }
 
   observe() {
-    if (this.observer) {
-      this.observer.observe(this.element, this.options);
+    if (DidResizeModifier.observer) {
+      this.addHandler();
+      DidResizeModifier.observer.observe(this.element, this.options);
     }
   }
 
   unobserve() {
-    if (this.observer) {
-      this.observer.unobserve();
+    if (DidResizeModifier.observer) {
+      DidResizeModifier.observer.unobserve(this.element);
     }
   }
 
   disconnect() {
-    if (this.observer) {
-      this.observer.disconnect();
+    if (DidResizeModifier.observer) {
+      DidResizeModifier.observer.disconnect();
```

Comment:
I think we want to avoid calling `disconnect` altogether, because that unobserves ALL elements, now that they share a single observer instance. 

https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver/disconnect

In reality, it seems we could eliminate this method entirely and just call `unobserve` where we're calling `disconnect` on line 86.